### PR TITLE
[wasm][testing] Avoid using crypto in WASM HTTP unit tests

### PR DESCRIPTION
**Pull request by pavelsavara**
On WASM platform existing POST/PUT tests are blocked by current [lack of crypto](https://github.com/dotnet/runtime/issues/37669) in WASM.
By skipping content checksum validation in the tests for Browser, we could enable about 130 tests of other HTTP features.

- send `Content-MD5-Skip` header instead of `Content-MD5` header from browser
- on the server side (hosted in XHarness) skip request MD5 validation when `Content-MD5-Skip` is present
- on the client side skip `Content-MD5` response validation
- `SendAsync_SendRequestUsingMethodToEchoServerWithContent_Success` more granular active issue https://github.com/dotnet/runtime/issues/53591
- fix for port handling on `SendAsync_GetWithValidHostHeader_Success`

### CODE REVIEW

File: src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs [hunk 1]
```diff
@@ -342,14 +339,23 @@ public async Task SendAsync_HttpRequestMsgResponseHeadersRead_StatusCodeOK(Confi
...
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/37669", TestPlatforms.Browser)]
         public async Task PostAsync_CallMethodTwice_StringContent(Configuration.Http.RemoteServer remoteServer)
         {
             using (HttpClient client = CreateHttpClientForRemoteServer(remoteServer))
             {
                 string data = "Test String";
                 var content = new StringContent(data, Encoding.UTF8);
-                content.Headers.ContentMD5 = TestHelper.ComputeMD5Hash(data);
+
+                if(PlatformDetection.IsBrowser)
```
Comment (ManickaP) [inline src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs:349]: NIT (same as other places)
```suggestion
                if (PlatformDetection.IsBrowser)
```

File: src/libraries/Common/tests/System/Net/Http/TestHelper.cs [hunk 2]
```diff
@@ -41,9 +41,13 @@ public static bool JsonMessageContainsKey(string message, string key)
             bool chunkedUpload,
             string requestBody)
         {
-            // Verify that response body from the server was corrected received by comparing MD5 hash.
-            byte[] actualMD5Hash = ComputeMD5Hash(responseContent);
-            Assert.Equal(expectedMD5Hash, actualMD5Hash);
+            // https://github.com/dotnet/runtime/issues/37669
```
Comment (ManickaP) [inline src/libraries/Common/tests/System/Net/Http/TestHelper.cs:44]:
```suggestion
            // [ActiveIssue("https://github.com/dotnet/runtime/issues/37669")
```

File: src/libraries/Common/tests/System/Net/Prerequisites/NetCoreServer/Handlers/VerifyUploadHandler.cs [hunk 3]
```diff
@@ -35,31 +35,40 @@ public static async Task InvokeAsync(HttpContext context)
...
-                }
-
-                if (expectedHash == actualHash)
+                string skip = context.Request.Headers["Content-MD5-Skip"];
+                if(!string.IsNullOrEmpty(skip)) 
                 {
                     context.Response.StatusCode = 200;
                 }
                 else
                 {
```
Comment (ManickaP) [inline src/libraries/Common/tests/System/Net/Prerequisites/NetCoreServer/Handlers/VerifyUploadHandler.cs:44]: We could get rid of one level of nesting:
```suggestion
                if(!string.IsNullOrEmpty(skip)) 
                {
                    context.Response.StatusCode = 200;
                    return;
                }
```
But this is thing of taste, so feel free to disregard, it's perfectly fine as-is as well.

**Review by ManickaP — APPROVED** [review]
LGTM, thanks!

File: src/libraries/Common/tests/System/Net/Prerequisites/NetCoreServer/Handlers/VerifyUploadHandler.cs [hunk 4]
```diff
@@ -32,40 +32,40 @@ public static async Task InvokeAsync(HttpContext context)
             // Get request body.
             byte[] requestBodyBytes = await ReadAllRequestBytesAsync(context);
 
-            // Check MD5 checksum for non-empty request body.
-            if (requestBodyBytes.Length > 0)
+            // Skip MD5 checksum for empty request body or for requests which opt to skip it.
```
Comment (radical) [inline src/libraries/Common/tests/System/Net/Prerequisites/NetCoreServer/Handlers/VerifyUploadHandler.cs:35]: nit: mention the issue here due to  which we are doing this.

**Review by radical — COMMENTED** [review]
Looks good, just a couple of comments.

**Review by radical — APPROVED** [review]

**Review by lewing — APPROVED** [review]

File: src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs [hunk 5]
```diff
@@ -358,7 +364,16 @@ public async Task PostAsync_CallMethodTwice_StringContent(Configuration.Http.Rem
 
                 // Repeat call.
                 content = new StringContent(data, Encoding.UTF8);
-                content.Headers.ContentMD5 = TestHelper.ComputeMD5Hash(data);
+                if (PlatformDetection.IsBrowser)
```
Comment (voloda) [inline src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.RemoteServer.cs:367]: Should you have this extracted to a method to reduce copy&paste? :-)

**Pull request merged**